sdk/core/AzureCore/Tests/QueryParametersTests.swift

```diff
@@ -0,0 +1,168 @@
+    case custom(String)
+    case square
+    case circle
+
+    var queryValue: String {
+        switch self {
+        case let .custom(val):
+            return val
+        case .square:
+            return "square"
+        case .circle:
+            return "circle"
+        }
+    }
+}
+
+typealias MyDate = Date
+extension MyDate: QueryStringConvertible {
+    internal static let dateFormat = "yyyy-MM-dd"
+
+    public var queryValue: String {
+        let formatter = DateFormatter()
+        formatter.dateFormat = MyDate.dateFormat
+        return formatter.string(from: self)
+    }
+}
+
+// MARK: Test Cases
+
+class QueryParametersTests: XCTestCase {
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    func test_QueryParameters_WithOptionals() throws {
+        let nilString: String? = nil
+        let query = QueryParameters([
+            ("var1", "test"),
+            ("var2", nilString)
+        ])
+        XCTAssert(query.parameters.count == 1)
+        XCTAssert(query.parameters[0].value == "test")
+    }
+
+    func test_QueryParameters_WithInt() throws {
+        let query = QueryParameters([
+            ("var1", 5)
+        ])
+        XCTAssert(query.parameters.count == 1)
+        XCTAssert(query.parameters[0].value == "5")
+    }
+
+    func test_QueryParameters_WithBool() throws {
+        let query = QueryParameters([
+            ("var1", true)
+        ])
+        XCTAssert(query.parameters.count == 1)
+        XCTAssert(query.parameters[0].value == "true")
+    }
+
+    func test_QueryParameters_WithDate() throws {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy/MM/dd"
+        let date = formatter.date(from: "2000/01/01")! as MyDate
+        let query = QueryParameters([
+            ("var1", date)
+        ])
+        XCTAssert(query.parameters.count == 1)
+        XCTAssert(query.parameters[0].value == "2000-01-01")
+    }
+
+    func test_QueryParameters_WithData() throws {
+        let data = "test".data(using: .utf8)
+        let query = QueryParameters([
+            ("var1", data)
+        ])
+        XCTAssert(query.parameters.count == 1)
+        XCTAssert(query.parameters[0].value == "test")
+    }
+
+    func test_QueryParameters_WithStringBackedEnum() throws {
+        let query = QueryParameters([
+            ("var1", AnimalEnum.cat)
+        ])
+        XCTAssert(query.parameters.count == 1)
+        XCTAssert(query.parameters[0].value == "cat")
+    }
+
+    func test_QueryParameters_WithExtensibleStringBackedEnum() throws {
+        let query = QueryParameters([
+            ("var1", ShapeEnum.custom("parallelogram")),
+            ("var2", ShapeEnum.circle)
+        ])
```

Comment:
An example of how the extensible enum would work. If someone wanted to use the value "custom", the would supply `.custom("custom")` so I think this implementation will suffice for all our purposes.